test/puppetca/puppetca_config_test.rb

```diff
@@ -4,7 +4,18 @@
 class PuppetCAConfigTest < Test::Unit::TestCase
   def test_omitted_settings_have_default_values
     Proxy::PuppetCa::Plugin.load_test_settings({})
-    assert_equal '/var/lib/puppet/ssl', Proxy::PuppetCa::Plugin.settings.ssldir
     assert_equal 'puppetca_hostname_whitelisting', Proxy::PuppetCa::Plugin.settings.use_provider
   end
+
+  def test_set_puppet_cert_ca_provider_for_4_2
```

Comment:
Could you copy this test and add `4.10.1` as well? Our installer uses the `puppetversion` fact which is a full version.